src/resolve/resolved-dnstls-openssl.c

```diff
@@ -14,6 +14,18 @@
 #include "resolved-dnstls.h"
 #include "resolved-manager.h"
 
+static char *dnstls_error_string(int ssl_error, char *buf, size_t count) {
+        if (ssl_error == SSL_ERROR_SSL)
+                ERR_error_string_n(ERR_get_error(), buf, count);
+        else
+                snprintf(buf, count, "SSL_get_error()=%d", ssl_error);
```

Comment:
so, do i get this right? openssl has an API to convert low-level crypto errors to friendly strings, but none to convert high-level SSL/TLS errors to strings? hmpf...

Reply:
That's right. Though in fairness, most of the values returned by `SSL_get_error` are some variation of "try again"